src/app/core/utils/api-token/api-token.service.ts

```diff
@@ -161,26 +185,39 @@ export class ApiTokenService {
     );
   }
 
+  getApiToken$(): Observable<string> {
+    return this.apiToken$.asObservable();
+  }
+
   /**
-   * Should remove the actual apiToken cookie and fetch a new anonymous user token
+   * remove the actual apiToken cookie
    */
   removeApiToken() {
     this.apiToken$.next(undefined);
   }
 
+  /**
+   * sets new apiToken and new apiToken cookie options when available
+   */
   setApiToken(apiToken: string, options?: CookieOptions) {
-    this.cookieOptions = options;
+    if (options) {
+      this.cookieOptions = options;
+    }
     this.apiToken$.next(apiToken);
   }
 
+  getCookieVanishes$(): Observable<ApiTokenCookieType> {
+    return this.cookieVanishes$.asObservable();
+  }
+
```

Comment:
I think this is not a good idea. The function is or can be used within the identity provider configurations. In case of the Auth0 identity provider the apiToken interception function is only used, when no idToken is available.